tune_sklearn/tune_search.py

```diff
@@ -306,17 +329,16 @@ def __init__(self,
             raise ValueError(
                 "Tune-sklearn no longer supports nested parallelism "
                 "with new versions of joblib/sklearn. Don't set 'sk_n_jobs'.")
-        search_optimization = search_optimization.lower()
-        available_optimizations = [
-            "random",
-            "bayesian",  # scikit-optimize/SkOpt
-            "bohb",
-            "hyperopt",
-            "optuna",
-        ]
-        if (search_optimization not in available_optimizations):
-            raise ValueError("Search optimization must be one of "
-                             f"{', '.join(available_optimizations)}")
+        if isinstance(search_optimization, str):
+            search_optimization = search_optimization.lower()
+
+        if (search_optimization not in set(available_optimizations.values())
+            ) and (not isinstance(search_optimization, type)
+                   or not issubclass(search_optimization, Searcher)):
+            raise ValueError(
+                "Search optimization must be one of "
+                f"{', '.join(list(available_optimizations.values()))} "
+                "or a ray.tune.suggest.Searcher class.")
```

Comment:
I couldn't figure out a good way to pass an already initialized object, as I have detailed in description - tl;dr there is no enforcement of attributes on Searcher subclassses, so while using objects would work with the Searchers present in Tune, it may not work with user-defined ones

Reply:
Hm, so Tune uses `set_search_properties()` to set the search space, metric, and mode of a search algorithm. The `Searcher` base class also has a `_metric` and `_mode` attribute. Can't we assume that custom searchers always implement that interface?
Custom searchers were not possible to use before with tune-sklearn, right? Thus, now would be a good time to define and enforce properties of these custom searchers

Reply:
I'd think that would need to be enforced in Ray Tune itself, but we can check it here for now. In that case, I'll make it so that an object is passed instead of a class